src/modules/payment/screens/payment-screen.tsx

```diff
@@ -22,8 +22,28 @@ export const PaymentScreen: React.FC = () => {
 
   const [checked, setChecked] = React.useState(true);
 
-  const onSubmitBuyHandler: BuyFormSubmitHandler = () => {
-    console.log(buyFormRef.current?.getValues());
+  const { mutate } = usePurchaseOrder({});
+
+  const onSubmitBuyHandler: BuyFormSubmitHandler = data => {
+    mutate({
+      order: {
+        credit_card: {
+          card_number: data.cardNumber,
+          exp_month: data.expirationMonth,
+          exp_year: data.expirationYear,
+          cvc: data.cvcCode,
+        },
+        shipping_address: {
+          city: data.city,
+          country: data.country,
+          line_1: data.adress1,
+          line_2: typeof data.adress2 === 'undefined' ? '-' : data.adress2,
+          postal_code: data.postalCode,
+          state: data.country,
+        },
+      },
+    });
+    navigation.navigate("Tab", {screen: "Home"});
```

Comment:
Consider moving this navigate method to the onSuccess callback in the `usePurchaseOrder`, with that you are sure that the user is going to the Home screen if the order was successfully placed.

Reply:
Done